File: cukcuk clone/app/src/main/java/vn/com/misa/cukcukstarterclone/data/IOnLoadedCallback.java
package vn.com.misa.cukcukstarterclone.data;

/**
 * - Mục đích Class: Interface Callback để xử lý dữ liệu khi dùng {@link LoadingAsyncTask}
 *
 * @created_by KhanhNQ on 09-Jan-21.
 */
public interface IOnLoadedCallback<T> {
    void onSuccess(T data);

    void onFailure(Exception e);
}

File: app/src/main/java/com/misa/fresher/data/IOnLoadedCallback.kt
package com.misa.fresher.data

/**
 * - interface's purpose:
 *
 * @author HTLong
 * @edit_at 2022-03-21
 */

interface IOnLoadedCallback<D, E> {
    fun onSuccess(data: D)
    fun onFailure(error: E)
    fun onException(e: Exception)
}